### ISSUE TITLE
Not an issue – just a question! (re ReplayGain)

### ISSUE THREAD

**Issue by DanielTheGreat** [stsaz/fmedia#92]
Does fmedia-GUI (Win7Pro64) action Replay Gain stored in FLAC tags?
If not by default, CAN it, and what settings must be changed to achieve this?

Thanks for a great, simple player, that works 'out-of-the-box', including sending DTS-encoded FLAC bitstreams to my external surround receiver without needing to make any special settings!

**Comment by stsaz** [comment]
Hi!

Currently ReplayGain isn't supported, but I think it shouldn't be too hard to implement it.

> If not already included, then I would change this to a feature request rather than 'just a question'!

Yes, let's make it a feature request!

*P.S. Although I agree this is an important feature, I usually use Dynamic Audio Normalizer filter instead, because it doesn't rely on meta information which can be missing or deceiving.  That's why I just thought that there's no point in supporting ReplayGain.*

**Comment by DanielTheGreat** [comment]
Simon,

Thanks for your response.

Regarding the use of Dynamic Audio Normalizer filter, I suspect it would change the data in a DTS audio stream, thus rendering the track as white noise on a surround receiver!  I've gone to a lot of trouble to (semi-automatically) 'scale' the audio in the DTS channels to a 'ReplayGain-equivalent' level before encoding them to DTS, so they sound the same volume as my ReplayGain-tagged stereo tracks.  My 30,000+ track playlist contains a mix of both stereo and DTS FLAC files, and I don't think that would work with Dynamic Audio Normalization!

For the time being I've switched to another Windows player (that uses ReplayGain), but if you can add this feature and fix a few other minor irritations, I would happily switch back and also make a small donation.

My other minor irritations are concerning the playlist display:
1. I can't find how to make it automatically scroll as each track finishes, so the currently-playing track is always near the middle of the displayed portion of the list.
2. When I close fmedia and later open it again, my playlist has gone.  I must re-load it and start all over again to find where I left off playing last time.  The player should remember this!

Regards,
Daniel
---------------
(Australia)

 

________________________________

**Comment by stsaz** [comment]
> My 30,000+ track playlist contains a mix of both stereo and DTS FLAC files, and I don't think that would work with Dynamic Audio Normalization!

Yeah, using DAN filter on the correctly tagged files with RG would be pointless.

> 1. I can't find how to make it automatically scroll as each track finishes,

Not implemented, but I agree it's a nice-to-have feature.

> 2. When I close fmedia and later open it again, my playlist has gone.

Is it on v1.31 (the latest version)?  Probably fmedia's zstd compression mechanism doesn't work correctly...  Please explore to fmedia directory, open `cmd` there and execute this:

```
fmedia --gui
```

and drag-n-drop any audio file into playlist view, then close the window.  What's printed in console?

If there's nothing, then exec this:

```
fmedia --gui -D
```

and repeat the same procedure and paste the output here.

**Comment by DanielTheGreat** [comment]
Simon,

Yes, it's v1.31, but it's proving me a liar! :-)

Now when I have a playlist (with one or more files), it DOES re-start with that list still there!  I see from your suggested process that it saves the playlist file:
C:\Program Files\fmedia>fmedia --gui
φfmedia v1.31 (win-amd64)
saved file C:\Users\Daniel\AppData\Roaming/fmedia/list1.m3uz, 0 kbytes

(Though I wonder how Windows handles the mix of fwd- and back-slashes!)

So my 'wishlist' is reduced to just these three features (the third is 'new' - I haven't mentioned it before):
1. Implement ReplayGain
2. Implement playlist auto-scrolling
3. Gapless playback (there's currently a tiny gap between the bitstreams of one song and the next).  This probably requires pre-buffering of the next song in the playlist.

Regards,
Daniel

 

________________________________
